packages/boba/contracts/contracts/bridges/L1NFTBridge.sol

```diff
@@ -123,10 +124,8 @@ contract L1NFTBridge is iL1NFTBridge, CrossDomainEnabled, ERC721Holder, Reentran
         address _l2NFTBridge
     )
         public
-        onlyOwner()
         initializer()
     {
-        require(messenger == address(0), "Contract has already been initialized.");
```

Comment:
nice! one question though around upgradeability is - if the initializer() modifier was present in the deployed contract from the beginning, then its safe to remove this messenger == address(0) check at this point. might need to verify

Reply:
hmmm... wouldn't the `initializer` modifier already cover that?
```
modifier initializer() {
        require(_initializing || _isConstructor() || !_initialized, "Initializable: contract is already initialized");

        bool isTopLevelCall = !_initializing;
        if (isTopLevelCall) {
            _initializing = true;
            _initialized = true;
        }

        _;

        if (isTopLevelCall) {
            _initializing = false;
        }
    }
```

or if you elaborate why is this important `was present in the deployed contract from the beginning,`?

Reply:
yes exactly, the initializer() modifier should cover that, but i was thinking if there's a chance this modifier wasn't present when the first time the deployed contract was initialized, the `_initialized` wouldn't be set, and if we later added that modifier in - we couldn't have called initialize a second time 
(although thats a rare chance because the upgradeable libraries needed to be initialized as well)

Reply:
based on our docs, we've deployed NFT bridges only to Rinkeby.